lstchain/image/modifier.py

```diff
@@ -0,0 +1,83 @@
+__all__ = [
+    'add_noise_in_pixels',
+    'smear_light_in_pixels',
+]
+
+import numpy as np
+
+def add_noise_in_pixels(rng, image, extra_noise_in_dim_pixels,
+                        extra_bias_in_dim_pixels, transition_charge,
+                        extra_noise_in_bright_pixels):
+    """
+
+    Parameters
+    ----------
+    rng: numpy.random.default_rng  random number generator
+
+    image: charges (p.e.) in the camera
+
+    To be tuned by comparing the starting MC and data:
+
+    extra_noise_in_dim_pixels: mean additional number of p.e. to be added (
+    Poisson noise) to pixels with charge below transition_charge
+
+    extra_bias_in_dim_pixels: mean bias (w.r.t. original charge) of the new
+    charge in pixels. Should be 0 for non-peak-search pulse integrators
+
+    transition_charge: border between "dim" and "bright" pixels
+
+    extra_noise_in_bright_pixels: mean additional number of p.e. to be added (
+    Poisson noise) to pixels with charge above transition_charge. This is
+    unbiased, i.e. Poisson noise is introduced, and its average subtracted,
+    so that the mean charge in bright pixels remains unaltered. This is
+    because we assume that above transition_charge the integration window
+    is determined by the Cherenkov light, and would not be modified by the
+    additional NSB noise (presumably small compared to the C-light)
+
+    Returns
+    -------
+    Modified (noisier) image
+
+    """
+
+    qcopy = image.copy()
```

Comment:
This can be simplified to:

```
bright_pixels = image > transition_charge
noise = np.where(bright_pixels, extra_noise_in_bright_pixels, extra_noise_in_dim_pixels)
bias = np.where(bright_pixels, -extra_noise_in_bright_pixels, extra_bias_in_dim_pixels - extra_noise_in_dim_pixels)

image = image + rng.poisson(noise) - bias
```

Reply:
Done

Reply:
And you make a copy of the image but modify the image itself. Is that intentional?